From 6de5727b82960dd4f0d861c45024f58af77d3660 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 21 Oct 2021 16:35:20 +0300
Subject: [PATCH] [SW-56687] gaudi2: pldm: skip image checksum

In order to expedite fit load process on pldm, this patch skips
checksum check for fit images.

Change-Id: Ia1eef82d0ac18cdee674311e9eec270c6884fd16
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 83e0d0b..7ca0343 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -17,6 +17,7 @@
 #include <pll.h>
 #include <uart.h>
 #include <pci.h>
+#include <sha.h>
 
 #define HBM_BAR_PF_NUM (1 << 4)
 
@@ -197,3 +198,9 @@
 {
 	return hl_is_debug_mode(HL_DEBUG_HW_PCIE);
 }
+
+void sha_csum(const unsigned char *input, unsigned int ilen,
+		unsigned char *output)
+{
+	sha384_csum(input, ilen, output, 0);
+}
diff --git a/arch/gaudi2/common/onix.c b/arch/gaudi2/common/onix.c
index 3f412d1..123c6a3 100644
--- a/arch/gaudi2/common/onix.c
+++ b/arch/gaudi2/common/onix.c
@@ -10,6 +10,7 @@
 #include <cmd_module.h>
 #include <uart.h>
 #include <memory.h>
+#include <sha.h>
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
@@ -98,3 +99,9 @@
 void preboot_update_bar_sizes(void) {}
 
 void hl_clock_gating_init(void) {}
+
+void sha_csum(const unsigned char *input, unsigned int ilen,
+		unsigned char *output)
+{
+	sha384_csum(input, ilen, output, 0);
+}
diff --git a/arch/gaudi2/common/pldm.c b/arch/gaudi2/common/pldm.c
index 9f930b1..3fbc047 100644
--- a/arch/gaudi2/common/pldm.c
+++ b/arch/gaudi2/common/pldm.c
@@ -70,3 +70,8 @@
 void preboot_init_pcie_phy(void) {}
 
 void preboot_update_bar_sizes(void) {}
+
+void sha_csum(const unsigned char *input, unsigned int ilen,
+		unsigned char *output)
+{
+}
diff --git a/arch/gaudi2/common/sha.c b/arch/gaudi2/common/sha.c
index ff56b4b..ee15314 100644
--- a/arch/gaudi2/common/sha.c
+++ b/arch/gaudi2/common/sha.c
@@ -12,12 +12,6 @@
 #include "hl.h"
 #include <sha.h>
 
-void sha_csum(const unsigned char *input, unsigned int ilen,
-		unsigned char *output)
-{
-	sha384_csum(input, ilen, output, 0);
-}
-
 void sha_csum_regions(const struct image_region region[], int region_count,
 			 u8 *output)
 {
